sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs

```diff
@@ -90,53 +92,80 @@ public override async ValueTask<AccessToken> GetTokenAsync(TokenRequestContext r
             return await GetTokenAsync(true, requestContext, cancellationToken).ConfigureAwait(false);
         }
 
-        private async Task<AccessToken> GetTokenAsync(bool isAsync, TokenRequestContext requestContext, CancellationToken cancellationToken)
+        private async ValueTask<AccessToken> GetTokenAsync(bool isAsync, TokenRequestContext requestContext, CancellationToken cancellationToken)
         {
             using CredentialDiagnosticScope scope = _pipeline.StartGetTokenScope("DefaultAzureCredential.GetToken", requestContext);
 
-            List<Exception> exceptions = new List<Exception>();
+            try
+            {
+                AccessToken token;
+
+                if (_credential != null)
+                {
+                    token = isAsync ? await _credential.GetTokenAsync(requestContext, cancellationToken).ConfigureAwait(false) : _credential.GetToken(requestContext, cancellationToken);
```

Comment:
Do we want to handle `CredentialUnavailableException` and look for a new source?

Reply:
I don't think so. I think it's easier to reason about the behavior if we pick a source and then don't change.